libs/sdk-core/src/invoice.rs

```diff
@@ -166,28 +171,35 @@ pub fn add_lsp_routing_hints(
     // We make sure the hint we add does not conflict with other hints.
     // The lsp hint takes priority so in case the lsp hop is already in one of the existing hints
     // We make sure not to include them in the new hints.
```

Comment:
```suggestion
    // We make sure the route hints we want to add do not conflict with invoice route hints.
    // When merging route hints, only route hints are added that go through different nodes than ones in the invoice route hints.
    // Otherwise when not merging route hints, the invoice route hints are replaced by the provided route hints.
```